airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -125,20 +128,46 @@ protected void acceptTracked(final AirbyteMessage message) throws Exception {
       final AirbyteRecordMessage recordMessage = message.getRecord();
       final AirbyteStreamNameNamespacePair stream = AirbyteStreamNameNamespacePair.fromRecordMessage(recordMessage);
 
+      // if stream is not part of list of streams to sync to then throw invalid stream exception
       if (!streamNames.contains(stream)) {
         throwUnrecognizedStream(catalog, message);
       }
 
+      // if the record is not valid then add to map of stream to number of records omitted, then short
+      // circuit
       if (!isValidRecord.apply(message.getRecord().getData())) {
         streamToIgnoredRecordCount.put(stream, streamToIgnoredRecordCount.getOrDefault(stream, 0L) + 1L);
         return;
       }
 
-      // if the buffer flushes, update the states appropriately.
-      // TODO: ryankfu (if true, this is where bundling up logic to also commit to airbyte_raw table)
-      if (bufferingStrategy.addRecord(stream, message)) {
-        markStatesAsFlushedToTmpDestination();
+      final Optional flushType = bufferingStrategy.addRecord(stream, message);
+      // if present means that a flush occurred
+      if (flushType.isPresent()) {
+        // TODO: (ryankfu) separate the logic for handling flush states in a separate
```

Comment:
"in a separate..."  - was there a continuation to the sentence?

Reply:
This comment was resolved with the changes within BigQuery. It was an internal note for myself that I didn't push to remote but was part of the BigQuery branch